torch_xla/csrc/tensor.cpp

```diff
@@ -442,10 +442,12 @@ xla::int64 XLATensor::GetNextTensorId() {
 }
 
 XLATensor::ViewIrNode XLATensor::GetViewIrNode(View* view) {
-  if (view->ir_node != nullptr && view->base_ir_node == view->alias->ir_node) {
+  if (view->ir_node != nullptr &&
+      view->ir_node->operand(0).node == view->alias->ir_node.get()) {
```

Comment:
Not really a problem with this patch, but is it correct to compare pointers? Can't we get spurious matches because of recycling addresses?

Reply:
Yes, in this case we are holding a shared pointer reference, so that pointer cannot be reused.